python/lsst/pipe/tasks/metrics.py

```diff
@@ -0,0 +1,230 @@
+    sources produced by deblending that were themselves deblended by later
+    processing. If sky source information is present, sky sources are excluded.
+
+    Notes
+    -----
+    The task excludes any non-sky sources in the catalog, but it does
+    not require that the catalog include a ``sky_sources`` column.
+    """
+    _DefaultName = "numDeblendChildSciSources"
+    ConfigClass = NumberDeblendChildSourcesMetricConfig
+
+    def run(self, sources):
+        """Count the number of science sources created by deblending.
+
+        Parameters
+        ----------
+        sources : `lsst.afw.table.SourceCatalog` or `None`
+            A science source catalog, which may be empty or `None`.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            A `~lsst.pipe.base.Struct` containing the following component:
+
+            ``measurement``
+                the total number of science sources from deblending
+                (`lsst.verify.Measurement`). If no deblending information is
+                available in ``sources``, this is `None`.
+
+        Raises
+        ------
+        MetricComputationError
+            Raised if ``sources`` is missing mandatory keys for
+            source catalogs.
+        """
+        if sources is None:
+            self.log.info("Nothing to do: no catalogs found.")
+            meas = None
+        elif "deblend_nChild" not in sources.schema:
+            self.log.info("Nothing to do: no deblending performed.")
+            meas = None
+        else:
+            try:
+                children = ((sources["parent"] != 0)            # deblend child
+                            & (sources["deblend_nChild"] == 0)  # not deblended
+                            )
+                children = _filterSkySources(sources, children)
+            except LookupError as e:
+                # Probably "parent"; all other columns already checked
+                raise MetricComputationError("Invalid input catalog") from e
+            else:
+                nChildren = np.count_nonzero(children)
+                meas = Measurement(self.config.metricName, nChildren * u.dimensionless_unscaled)
+
+        return Struct(measurement=meas)
+
+
+def _filterSkySources(catalog, selection):
+    """Filter out any sky sources from a vector of selected sources.
+
+    If no sky source information is available, all sources are assumed to
+    be non-sky.
+
+    Parameters
+    ----------
+    catalog : `lsst.afw.table.SourceCatalog`
+        The catalog to filter.
+    selection : `numpy.ndarray` [`bool`], (N,)
+        A vector of existing source selections, of the same length as
+        ``catalog``, where selected sources are marked `True`.
+
+    Returns
+    -------
+    filtered : `numpy.ndarray` [`bool`], (N,)
+        A version of ``selection`` with any sky sources filtered out
+        (set to `False`). May be the same vector as ``selection`` if
+        no changes were made.
```

Comment:
Catching up, sorry - "selection" is fine, this docstring makes it pretty clear what is happening 👍